FormalConjectures/Wikipedia/CollatzConjecture.lean

```diff
@@ -0,0 +1,28 @@
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Collatz conjecture
+
+*Reference:* [Wikipedia](https://en.wikipedia.org/wiki/Collatz_conjecture)
+-/
+/--
+Consider the following operation on an arbitrary positive integer:
+If the number is even, divide it by two.
+If the number is odd, triple it and add one.
+Formally, we define the function f as follows:
+-/
+def f (n : ℕ) : ℕ :=
+  if n % 2 = 0 then n / 2 else 3 * n + 1
+
+/--
+Now form a sequence by performing this operation repeatedly, beginning with any positive integer, and taking the result at each step as the input at the next.
+Formally:
+-/
+def collatzSeq (n i : ℕ) : ℕ :=
```

Comment:
I would not define this separately, and instead use `Nat.iterate` in the main statement for iterate applications of `f`